api/v1beta2/awsmachine_types.go

```diff
@@ -153,6 +163,11 @@ type AWSMachineSpec struct {
 	// +kubebuilder:validation:MaxItems=2
 	NetworkInterfaces []string `json:"networkInterfaces,omitempty"`
 
+	// NetworkInterfaceType is the interface type of the primary network Interface.
```

Comment:
Maybe mention here that the default is `interface` and that `""` and `interface` are equivalent.

Reply:
Maybe we can have explicit default here? E.g. `// +kubebuilder:default=interface`

Reply:
Is it possible in the future that we might want to change the default? For example, what if Amazon deprecates the current interface type and makes EFA the choice, we would want to move folks across to EFA. If we add an explicit default here, then they would have to manually update their configs to move across should things ever change

Reply:
Added note that when field is not set, aws defaults the value.